### PR TITLE
Setup associations

### PR DESCRIPTION
**Pull request by hopeforkoami**
In this branch, I :

  - Create a class `Classroom `with the following:
     *` @label` instance variable, should be initialized in the constructor.
     * Setter and getter for `@label `(remember about `attr_accessor`).
  - Create the `has-many`/`belongs`-to relationship between `Classroom `and `Student.` The following should be implemented:
      * Create the `has-many `side (a classroom has many students).
      * Create the `belongs-to` side (a student belongs to a classroom).
      * Make sure that when adding a student to a classroom it also sets the classroom for the student.
      * Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
  - Create a class `Book` with the following:
      * `@title` and `@author` instance variables, should be initialized in the constructor.
      * Setters and getters for instance variables (remember about `attr_accessor`).
  - Create a class `Rental` with the following:
      * `@date` instance variable, should be initialized in the constructor.
      * Setter and getter for `@date` (remember about `attr_accessor`).
  - Create the `many-to-many` (also` has-many-through`) relationship between `Person` and `Book` using the intermediate class Rental. The following should be implemented:
      * Create the `has-many` side of` Book` and `Rental `(a book has many rentals).
      * Create the `belongs-to` side of `Rental` and `Book `(a rental belongs to a book).
      * Create the `has-many` side of `Person `and `Rental `(a person has many rentals).
      * Create the `belongs-to` side of `Rental `and `Person `(a rental belongs to a person).
      * Modify the constructor of `Rental `so `Book `and `Person `are set in it.
      
Thanks

### CODE REVIEW

File: entity/book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (Mihndim2020) [inline entity/book.rb:9]: - [ ] To implement the `has-many` side of the relationship(`association`) between book and rental, a book has many rentals, a method for adding rentals should be present in the Book class. Something like so: 

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

Please kindly fix that.

File: entity/person.rb [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name, parent_permission: true)
     super
     @id = Random.rand(1..10_000)
     @name = name
     @age = age
     @permission = parent_permission
+    @rentals = []
   end
```
Comment (Mihndim2020) [inline entity/person.rb:15]: - [ ] To implement the `has-many` side of the relationship(`association`) between person and rental, a person has many rentals, a method for adding rentals should be present in the Person class. Something like so: 

```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

Please kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @hopeforkoami, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and Organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @hopeforkoami , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

200w


## Highlights
Good job so far :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**